Finmer.Core/Serialization/FurballContentWriterBinary.cs

```diff
@@ -84,11 +98,14 @@ public void WriteByteArrayProperty(string key, byte[] value)
         {
             if (value == null)
             {
-                m_Stream.Write(-1);
+                WriteCompressedInt32Property(null, 0);
             }
             else
             {
-                m_Stream.Write(value.Length);
+                if (value.Length == 0)
```

Comment:
The reason I made this throw is because otherwise a zero-length array will be read as `null` later, making the serialization code somewhat asymmetric. The only case where this could happen is with weird empty item icons (which would be re-interpreted as whatever a `null` icon attachment means to the editor -- maybe no item icon when unpacked.)

I don't imagine this case is likely to ever get hit, unless the byte array methods get used for things other than attachments (although a byte array in JSON seems pretty ugly).

As for why it *should* throw, mostly to avoid the aforementioned asymmetry, although I guess in this case the distinction between `null` and zero length doesn't really need to exist.

Maybe a better solution would be to move the byte array stuff into the attachment methods, remove the byte array methods, and add a comment to `AssetItem`? (And remove this exception of course.) This would mean no one can write arbitrary byte arrays anymore, but I can't see any use for those outside of attachments anyway.